storage-node-v2/src/services/webApi/controllers/stateApi.ts

```diff
@@ -0,0 +1,157 @@
+import { getDataObjectIDs } from '../../../services/caching/localDataObjects'
+import * as express from 'express'
+import _ from 'lodash'
+import { getDataObjectIDsByBagId } from '../../sync/storageObligations'
+import { WebApiError, sendResponseWithError, AppConfig } from './common'
+import fastFolderSize from 'fast-folder-size'
+import { promisify } from 'util'
+import fs from 'fs'
+import NodeCache from 'node-cache'
+const fsPromises = fs.promises
+
+// Expiration period in seconds for the local cache.
+const ExpirationPeriod = 30
+
+// Local in-memory cache for data
+const dataCache = new NodeCache({
+  stdTTL: ExpirationPeriod,
+  deleteOnExpire: true,
+})
+
+/**
+ * A public endpoint: return all local data objects.
+ */
+export async function getAllLocalDataObjects(
+  req: express.Request,
+  res: express.Response<unknown, AppConfig>
+): Promise<void> {
+  try {
+    const ids = await getDataObjectIDs()
+
+    res.status(200).json(ids)
+  } catch (err) {
+    sendResponseWithError(res, err, 'all_data_objects')
+  }
+}
+
+/**
+ * A public endpoint: serves local data uploading directory stats.
+ *
+ *  @return total size and count of the data objects.
+ */
+export async function getLocalDataStats(
+  req: express.Request,
+  res: express.Response<unknown, AppConfig>
+): Promise<void> {
+  try {
+    const uploadsDir = res.locals.uploadsDir
+    const tempFileDir = res.locals.tempFileUploadingDir
+    const fastFolderSizeAsync = promisify(fastFolderSize)
+
+    const tempFolderExists = fs.existsSync(tempFileDir)
+    const statsPromise = fsPromises.readdir(uploadsDir)
+    const sizePromise = fastFolderSizeAsync(uploadsDir)
+
+    const [stats, totalSize] = await Promise.all([statsPromise, sizePromise])
+
+    let objectNumber = stats.length
+    let tempDownloads = 0
+    let tempDirSize = 0
+    if (tempFolderExists) {
+      if (objectNumber > 0) {
+        objectNumber--
+      }
+
+      const tempDirStatsPromise = fsPromises.readdir(tempFileDir)
+      const tempDirSizePromise = fastFolderSizeAsync(tempFileDir)
+
+      const [tempDirStats, tempSize] = await Promise.all([tempDirStatsPromise, tempDirSizePromise])
+
+      tempDirSize = tempSize ?? 0
+      tempDownloads = tempDirStats.length
+    }
+
+    res.status(200).json({
+      objectNumber,
+      totalSize,
+      tempDownloads,
+      tempDirSize,
+    })
+  } catch (err) {
+    sendResponseWithError(res, err, 'local_data_stats')
+  }
+}
+
+/**
+ * A public endpoint: return local data objects for the bag.
+ */
+export async function getLocalDataObjectsByBagId(
+  req: express.Request,
+  res: express.Response<unknown, AppConfig>
+): Promise<void> {
+  try {
+    const queryNodeUrl = res.locals.queryNodeEndpoint
+    const bagId = getBagId(req)
+
+    const [ids, requiredIds] = await Promise.all([
+      getDataObjectIDs(),
+      getCachedDataObjectsObligations(queryNodeUrl, bagId),
```

Comment:
Implementation of `getCachedDataObjectsObligations()` does not use bagId when storing cached data so wrong cache results are returned for different bags.

Reply:
Fixed